solr/core/src/test/org/apache/solr/schema/TestCloudManagedSchema.java

```diff
@@ -53,12 +54,11 @@ public void test() throws Exception {
     QueryRequest request = new QueryRequest(params);
     request.setPath("/admin/cores");
     int which = r.nextInt(clients.size());
-    HttpSolrClient client = (HttpSolrClient) clients.get(which);
-    String previousBaseURL = client.getBaseURL();
-    // Strip /collection1 step from baseURL - requests fail otherwise
-    client.setBaseURL(previousBaseURL.substring(0, previousBaseURL.lastIndexOf("/")));
-    NamedList<?> namedListResponse = client.request(request);
-    client.setBaseURL(previousBaseURL); // Restore baseURL
+
+    // create a client that does not have the /collection1 as part of the URL.
+    SolrClient rootClient =
+        new HttpSolrClient.Builder(buildUrl(jettys.get(which).getLocalPort())).build();
+    NamedList<?> namedListResponse = rootClient.request(request);
```

Comment:
Your change will probably leave an un-closed SolrClient hanging around, which will fail.

Reply:
Done, and did the same with the `zkClient`